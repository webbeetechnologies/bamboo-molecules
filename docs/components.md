# Bamboo Molecules

Bamboo Molecules components are built on top of BambooAtoms; thus, at it's core; Bamboo allows for all the components to be composed by default. That means; you can replace an underlying view by a Styled View of your choice; and that is only the beginning.


Bamboo atoms have minimal theming; and Bamboo Molecules are opinionated styles; however, Molecules allow you to customize or entirely change the theme of a specific component. It is possible to use custom components/styles at a module level.


Molecules is heavily inspired by Material and Cupertino components and thus tries to mimic them to provide a native experience as closely as possible on the react native platform.



# References:
- [Comparing iOS and Android UI](https://www.learnui.design/blog/ios-vs-android-app-ui-design-complete-guide.html)
- [Material 3 Figma Design](https://www.figma.com/file/hGlWM0wnP1UN5VtPGiaOJ2/Material-3-Design-Kit-(Community)?node-id=50716%3A11360)
- [Material 3 Design Guidelines](https://m3.material.io/components)

## Molecules HOC:
- [WithAddons](./components/HOC/WithAddons.md)
- [WithAddonAfter](./components/HOC/WithAddons.md#withaddonafter)
- [WithAddonBefore](./components/HOC/WithAddons.md#withaddonbefore)
- [WithElements](./components/HOC/WithElements.md)
- [WithActionIconButton](./components/HOC/WithActionIconButton.md)
- [WithPopover](./components/HOC/WithPopover.md)


## Molecules created using Atoms:
- High Priority
    -   [Icons](./components/Icons.md)
    -   [Input](./components/Input.md)
    -   [Tooltips](./components/Tooltips.md)
    -   Actionsheet
    -   [Flat List](./components/FlatList.md)
    -   Modals/ Dialogs
    -   [Button](./components/Button.md)
    -   [IconButton](./components/IconButton.md)
    -   [Select](./components/Select.md)
    -   [Radio](./components/Radio.md)
    -   [Checkbox](./components/Checkbox.md)
    -   [Date Picker](./components/DatePicker.md)
    -   [Time Picker](./components/TimePicker.md)
    -   Menu and Menu Items
    -   Alert
    -   Cards
    -   Accordion
    -   [Chips](./components/Chips.md)
    -   Drawer


- Medium Priority
    -   [Section List](./components/SectionList.md)
    -   [Range Picker](./components/RangePickers.md)
    -   [Date Time Picker](./components/DateTimePicker.md)
    -   Stack
    -   Divider
    -   Toast
    -   Fab/ Fab Group (Android/Web Only)
    -   Tabs
        -   Pills
        -   Bottom Tabs


- Low Priority
    -   [Color Picker](./components/ColorPicker.md)
    -   [FilePicker](./components/FilePicker.md)
    -   Badges
    -   Search
    -   Forms
    -   Avatar
    -   Graphs




... and more as we identify ..
