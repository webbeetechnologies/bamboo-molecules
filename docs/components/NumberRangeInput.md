# Number Range

Number ranges can be open ended, and sometimes they have a maximum and minimum values for selection. Thus, molecules takes this into accounts and implements the Number Range component.

## Explanation Video
-   [Explanining Number Range](https://www.loom.com/share/59bb6ac407cc4b948aa883bec10f86f6)

## References
-   [Butlerapp](https://bamboo.veranstaltungsbutler.de/)

## Personas
-   End User
-   Component Consumer


### End User
-   As an end user, I want to be able to manually enter values numeric values in text boxes.
-   As an end user, I expect to be able to enter only numeric values into the input.
-   As an end user, I expect to see a numeric keypad on mobile/tablet platforms.
-   As an end user, I want to be able to type in decimal numbers.
-   As an end user, I want to see validation messages when the `minValue` value is greater than `maxValue` value entered.
-   As an end user, I expect that the validation messages are only shown once I am done entering the values.
-   As an end user, I want to be able to enter same values for min and max input fields.


### Component Consumer
-   As a component consumer, I expect that the component implements Number Input Components.
-   As a component consumer, I want may not always be able to define a `minValue` and `maxValue` value for the number component
-   As a component consumer, I expect the user input is validated against `minValue` and `maxValue` props as available.
-   As a component consumer, I want to be able to be able to listen into the onChange events.


## Interfaces
```ts
    interface NumberRangeProps {
        value: { min: number, max: number },
        minValue?: number;
        maxValue?: number;
    }
```