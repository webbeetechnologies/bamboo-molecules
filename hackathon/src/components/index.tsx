export { default as AttributeValueInput } from './AttributeValueInput/AttributeValueInput';
export { default as CustomDrawer } from './CustomDrawer/CustomDrawer';
export { default as MachineCard } from './MachineCard/MachineCard';
export { default as MachineTypeCard } from './MachineTypeCard/MachineTypeCard';
