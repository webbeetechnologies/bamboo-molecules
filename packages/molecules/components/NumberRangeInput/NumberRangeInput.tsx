import { memo, useCallback, useEffect, useMemo, useState } from 'react';
import { StyleProp, View, ViewStyle, type ViewProps } from 'react-native';
import { useControlledValue } from '../../hooks';
import { NumberInput, type NumberInputProps } from '../NumberInput';
import { isNil } from '../../utils';
import { ElementGroup } from '../ElementGroup';
import { HelperText } from '../HelperText';
import { numberRangeInputStyles } from './utils';

type Value = {
    min: string;
    max: string;
};

export type Props = ViewProps & {
    inputsContainerStyle?: ViewProps;
    value: Value;
    onChange: (args: { min: string; max: string }) => void;
    minInputProps?: Omit<NumberInputProps, 'variant' | 'value' | 'onChangeText' | 'onChange'>;
    maxInputProps?: Omit<NumberInputProps, 'variant' | 'value' | 'onChangeText' | 'onChange'>;
    variant?: NumberInputProps['variant'];
    errorMessage?: string;
};

const NumberRangeInput = ({
    variant = 'flat',
    inputsContainerStyle: inputsContainerStyleProp = {},
    minInputProps: {
        style: minInputStyleProp = {},
        onFocus: onFocusMinInput,
        onBlur: onBlurMinInput,
        ...minInputProps
    } = {},
    maxInputProps: {
        style: maxInputStyleProp = {},
        onFocus: onFocusMaxInput,
        onBlur: onBlurMaxInput,
        ...maxInputProps
    } = {},
    value: valueProp,
    onChange,
    errorMessage = 'Invalid number range.',
    style,
    ...rest
}: Props) => {
    const [value, onValueChange] = useControlledValue({
        value: valueProp,
        onChange,
    });
    const [error, setError] = useState(false);
    const [isTouched, setIsTouched] = useState(false); // TODO - create a hook

    const { containerStyle, inputsContainerStyle, minInputStyle, maxInputStyle, errorTextStyle } =
        useMemo(() => {
            const { minInput, maxInput, errorText } = numberRangeInputStyles;

            return {
                containerStyle: [numberRangeInputStyles.root, style],
                inputsContainerStyle: inputsContainerStyleProp,
                minInputStyle: [minInput, minInputStyleProp] as StyleProp<ViewStyle>,
                maxInputStyle: [maxInput, maxInputStyleProp],
                errorTextStyle: errorText,
            };
        }, [inputsContainerStyleProp, maxInputStyleProp, minInputStyleProp, style]);

    const onInputChange = useCallback(
        (type: 'min' | 'max', text: string) => {
            const newValue = {
                ...(value || {}),
                [type]: text,
            };

            onValueChange(newValue);
        },
        [onValueChange, value],
    );

    const onFocus = useCallback(
        (args: any, type: 'min' | 'max') => {
            if (!isTouched) setIsTouched(true);

            if (type === 'min') {
                onFocusMinInput?.(args);
                return;
            }

            onFocusMaxInput?.(args);
        },
        [isTouched, onFocusMaxInput, onFocusMinInput],
    );

    const onBlur = useCallback(
        (args: any, type: 'min' | 'max') => {
            if (type === 'min') {
                onBlurMinInput?.(args);
                return;
            }

            onBlurMaxInput?.(args);
        },
        [onBlurMaxInput, onBlurMinInput],
    );

    useEffect(() => {
        if (
            !isTouched ||
            value?.min === '' ||
            value?.max === '' ||
            isNil(value?.min) ||
            isNil(value?.max)
        )
            return;

        setError(Number(value?.min) > Number(value?.max));
    }, [isTouched, value?.max, value?.min]);

    return (
        <View style={containerStyle}>
            <ElementGroup style={inputsContainerStyle} {...rest}>
                <NumberInput
                    label="min"
                    {...minInputProps}
                    variant={variant}
                    containerStyle={minInputStyle}
                    value={value?.min}
                    onChangeText={useCallback(
                        (text: string) => onInputChange('min', text),
                        [onInputChange],
                    )}
                    onFocus={useCallback((args: any) => onFocus(args, 'min'), [onFocus])}
                    onBlur={useCallback((args: any) => onBlur(args, 'min'), [onBlur])}
                />
                <NumberInput
                    label="max"
                    {...maxInputProps}
                    variant={variant}
                    containerStyle={maxInputStyle}
                    value={value?.max}
                    onChangeText={useCallback(
                        (text: string) => onInputChange('max', text),
                        [onInputChange],
                    )}
                    onFocus={useCallback((args: any) => onFocus(args, 'max'), [onFocus])}
                    onBlur={useCallback((args: any) => onBlur(args, 'max'), [onBlur])}
                />
            </ElementGroup>
            {error && <HelperText style={errorTextStyle}>{errorMessage}</HelperText>}
        </View>
    );
};

export default memo(NumberRangeInput);
