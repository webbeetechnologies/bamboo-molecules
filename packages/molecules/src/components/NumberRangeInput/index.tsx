import { getRegisteredMoleculesComponent, registerMoleculesComponents } from '../../core';
import NumberRangeInputDefault from './NumberRangeInput';

registerMoleculesComponents({
    NumberRangeInput: NumberRangeInputDefault,
});

export const NumberRangeInput = getRegisteredMoleculesComponent('NumberRangeInput');

export { numberRangeInputStyles } from './utils';
