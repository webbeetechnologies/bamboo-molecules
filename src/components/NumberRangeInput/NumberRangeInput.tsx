import { memo, useCallback, useEffect, useMemo, useState } from 'react';
import type { ViewProps } from '@webbee/bamboo-atoms';
import { useComponentStyles, useControlledValue, useMolecules } from '../../hooks';
import type { NumberInputProps } from '../NumberInput';
import { isNil } from '../../utils';

type Value = {
    min: string;
    max: string;
};

export type Props = ViewProps & {
    inputsContainerStyle?: ViewProps;
    dividerStyle?: ViewProps;
    value?: Value;
    defaultValue?: Value;
    onChange?: (args: { min: string; max: string }) => void;
    minInputProps?: Omit<NumberInputProps, 'variant' | 'value' | 'onChangeText' | 'onChange'>;
    maxInputProps?: Omit<NumberInputProps, 'variant' | 'value' | 'onChangeText' | 'onChange'>;
    variant?: NumberInputProps['variant'];
    errorMessage?: string;
};

const NumberRangeInput = ({
    variant = 'flat',
    inputsContainerStyle: inputsContainerStyleProp = {},
    dividerStyle: dividerStyleProp = {},
    minInputProps: {
        style: minInputStyleProp = {},
        onFocus: onFocusMinInput,
        onBlur: onBlurMinInput,
        ...minInputProps
    } = {},
    maxInputProps: {
        style: maxInputStyleProp = {},
        onFocus: onFocusMaxInput,
        onBlur: onBlurMaxInput,
        ...maxInputProps
    } = {},
    value: valueProp,
    defaultValue,
    onChange,
    errorMessage = 'Invalid number range.',
    style,
    ...rest
}: Props) => {
    const { View, ElementGroup, NumberInput, HelperText } = useMolecules();
    const componentStyles = useComponentStyles('NumberRangeInput', [
        style,
        {
            inputsContainer: inputsContainerStyleProp,
            minInput: minInputStyleProp,
            maxInput: maxInputStyleProp,
            divider: dividerStyleProp,
        },
    ]);
    const [value, onValueChange] = useControlledValue({
        value: valueProp,
        defaultValue,
        onChange,
    });
    const [error, setError] = useState(false);
    const [isTouched, setIsTouched] = useState(false); // TODO - create a hook

    const { containerStyle, inputsContainerStyle, minInputStyle, maxInputStyle, errorTextStyle } =
        useMemo(() => {
            const { inputsContainer, minInput, maxInput, errorText, ...restStyle } =
                componentStyles;

            return {
                containerStyle: restStyle,
                inputsContainerStyle: inputsContainer,
                minInputStyle: minInput,
                maxInputStyle: maxInput,
                errorTextStyle: errorText,
            };
        }, [componentStyles]);

    const onInputChange = useCallback(
        (type: 'min' | 'max', text: string) => {
            const newValue = {
                ...(value || {}),
                [type]: text,
            };

            onValueChange(newValue);
        },
        [onValueChange, value],
    );

    const onFocus = useCallback(
        (args: any, type: 'min' | 'max') => {
            if (!isTouched) setIsTouched(true);

            if (type === 'min') {
                onFocusMinInput?.(args);
                return;
            }

            onFocusMaxInput?.(args);
        },
        [isTouched, onFocusMaxInput, onFocusMinInput],
    );

    const onBlur = useCallback(
        (args: any, type: 'min' | 'max') => {
            if (type === 'min') {
                onBlurMinInput?.(args);
                return;
            }

            onBlurMaxInput?.(args);
        },
        [onBlurMaxInput, onBlurMinInput],
    );

    useEffect(() => {
        if (
            !isTouched ||
            value?.min === '' ||
            value?.max === '' ||
            isNil(value?.min) ||
            isNil(value?.max)
        )
            return;

        setError(Number(value?.min) > Number(value?.max));
    }, [isTouched, value?.max, value?.min]);

    return (
        <View style={containerStyle}>
            <ElementGroup style={inputsContainerStyle} {...rest}>
                <NumberInput
                    label="min"
                    {...minInputProps}
                    variant={variant}
                    containerStyle={minInputStyle}
                    value={value?.min}
                    onChangeText={text => onInputChange('min', text)}
                    onFocus={(args: any) => onFocus(args, 'min')}
                    onBlur={(args: any) => onBlur(args, 'min')}
                />
                <NumberInput
                    label="max"
                    {...maxInputProps}
                    variant={variant}
                    containerStyle={maxInputStyle}
                    value={value?.max}
                    onChangeText={text => onInputChange('max', text)}
                    onFocus={(args: any) => onFocus(args, 'max')}
                    onBlur={(args: any) => onBlur(args, 'max')}
                />
            </ElementGroup>
            {error && <HelperText style={errorTextStyle}>{errorMessage}</HelperText>}
        </View>
    );
};

export default memo(NumberRangeInput);
