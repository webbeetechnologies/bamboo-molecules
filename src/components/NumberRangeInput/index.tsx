export { default as NumberRangeInput, Props as NumberRangeInputProps } from './NumberRangeInput';
export { numberRangeInputStyles } from './utils';
