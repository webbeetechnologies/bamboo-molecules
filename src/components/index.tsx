export {
    TextProps,
    HeadingProps,
    H1Props,
    H2Props,
    H3Props,
    H4Props,
    H6Props,
    UnderlineProps,
    StrongProps,
    LabelProps,
    StrikethroughProps,
    ItalicProps,
    ViewProps,
} from '@webbee/bamboo-atoms';

export { Icon, registerCustomIconType, IconProps, iconStyles } from './Icon';
export { TouchableRipple, TouchableRippleProps, touchableRippleStyles } from './TouchableRipple';
export {
    ActivityIndicator,
    ActivityIndicatorProps,
    activityIndicatorStyles,
} from './ActivityIndicator';
export { Surface, SurfaceProps, surfaceStyles } from './Surface';
export { Button, ButtonProps, buttonStyles } from './Button';
export {
    HorizontalDivider,
    horizontalDividerStyles,
    HorizontalDividerProps,
} from './HorizontalDivider';
export { VerticalDivider, verticalDividerStyles, VerticalDividerProps } from './VerticalDivider';
export { Switch, switchStyles, SwitchProps } from './Switch';
export { FlatList, FlatListProps, IFlatList, flatListStyles } from './FlatList';

export {
    SectionList,
    SectionListProps,
    ISectionList,
    SectionListRenderItem,
    SectionListRenderItemInfo,
    DefaultSectionT,
    sectionListStyles,
} from './SectionList';

export {
    default as ListItem,
    ListItemProps,
    listItemStyles,
    listItemTitleStyles,
    listItemDescriptionStyles,
} from './ListItem';

export { FlatGrid, IFlatGrid, flatGridStyles, FlatGridProps } from './FlatGrid';

export { SectionGrid, ISectionGrid, sectionGridStyles, SectionGridProps } from './SectionGrid';

export { Popover, PopoverProps, popoverStyles } from './Popover';
export { Overlay, OverlayProps } from './Overlay';
export { Backdrop, BackdropProps, backdropStyles } from './Backdrop';
export * from './Animations';

export { IconButton, IconButtonProps, iconButtonStyles } from './IconButton';
export { default as Checkbox, CheckboxProps, checkboxStyles } from './Checkbox';
export { TextInput, TextInputProps, textInputStyles } from './TextInput';
export { NumberInput, NumberInputProps } from './NumberInput';
export { FilePicker, FilePickerProps } from './FilePicker';
export { HelperText, HelperTextProps, helperTextStyles } from './HelperText';

export { Modal, ModalProps, modalStyles } from './Modal';

export {
    DatePickerInline,
    DatePickerInlineProps,
    datePickerStyles,
    datePickerYearPickerStyles,
    datePickerMonthStyles,
    datePickerDayStyles,
    datePickerHeaderStyles,
    dateDayNameStyles,
    datePickerDayRangeStyles,
    datePickerDayEmptyStyles,
    datePickerWeekStyles,
    CalendarDate,
    CalendarDates,
} from './DatePickerInline';

export {
    DatePickerModal,
    DatePickerModalProps,
    datePickerModalStyles,
    datePickerModalContentHeaderStyles,
    datePickerModalHeaderStyles,
    datePickerModalHeaderBackgroundStyles,
    datePickerModalEditStyles,
} from './DatePickerModal';

export { DatePickerInput, DatePickerInputProps, datePickerInputStyles } from './DatePickerInput';

export {
    TimePicker,
    TimePickerProps,
    timePickerStyles,
    timePickerInputStyles,
    timePickerInputsStyles,
    timePickerClockStyles,
    timePickerClockHoursStyles,
    timePickerClockMinutesStyles,
    timePickerAmPmSwitcherStyles,
} from './TimePicker';

export { TimePickerModal, TimePickerModalProps, timePickerModalStyles } from './TimePickerModal';

export { TextInputWithMask, TextInputWithMaskProps } from './TextInputWithMask';

export { OptionList, OptionListProps, IOptionList, optionListStyles } from './OptionList';

export { DropdownList, DropdownListProps, IDropdownList, dropdownListStyles } from './DropdownList';

export { ElementGroup, elementGroupStyles, ElementGroupProps } from './ElementGroup';

export { InputAddon, inputAddonStyles, InputAddonProps } from './InputAddon';

export * from './ActionSheet';

export * from './Dialog';

export * from './RadioButton';

export * from './NumberRangeInput';

export {
    ProgressCircle,
    ProgressCircleProps,
    progressCircleStyles,
} from './ProgressIndicator/Circle';
export { ProgressBar, ProgressBarProps, progressBarStyles } from './ProgressIndicator/Bar';

export * from './DropdownListPopover';

export * from './Card';

export * from './Select';

export * from './Chip';
