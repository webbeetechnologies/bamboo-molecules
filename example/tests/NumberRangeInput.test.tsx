import { renderWithWrapper } from '../testHelpers';
import { NumberRangeInput } from '../../src/components/NumberRangeInput';

const mockedOnChange = jest.fn();

describe('NumberRangeInput', () => {
    it('should render correctly with default props', () => {
        const tree = renderWithWrapper(<NumberRangeInput />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should correctly mask the defaultValue', () => {
        const { getByDisplayValue } = renderWithWrapper(
            <NumberRangeInput defaultValue={{ min: '10', max: '15' }} />,
        );
        expect(getByDisplayValue('10')).toBeTruthy();
        expect(getByDisplayValue('15')).toBeTruthy();
    });

    it('should correctly work with value and onChange', () => {
        const { getByDisplayValue } = renderWithWrapper(
            <NumberRangeInput value={{ min: '10', max: '15' }} onChange={mockedOnChange} />,
        );
        expect(getByDisplayValue('10')).toBeTruthy();
        expect(getByDisplayValue('15')).toBeTruthy();
    });
});
