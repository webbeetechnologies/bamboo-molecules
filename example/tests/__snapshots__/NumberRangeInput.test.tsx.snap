// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberRangeInput should render correctly with default props 1`] = `
<View
  style={
    Object {
      "divider": Object {},
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "animationScale": 1,
            "backgroundColor": "rgba(231, 224, 236, 1)",
            "borderBottomRightRadius": 0,
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 0,
            "flexDirection": "row",
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "rgba(73, 69, 79, 1)",
              "bottom": 0,
              "height": 1,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "zIndex": 1,
            }
          }
          testID="text-input-underline"
        />
        <View
          style={
            Object {
              "flexGrow": 1,
              "minHeight": 56,
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <View
            collapsable={false}
            onLayout={[Function]}
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "zIndex": 3,
              }
            }
          >
            <Text
              collapsable={false}
              maxFontSizeMultiplier={15}
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "left": 0,
                  "lineHeight": 24,
                  "opacity": 0,
                  "position": "absolute",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
              testID="text-input-label-active"
            >
              min
            </Text>
            <Text
              collapsable={false}
              maxFontSizeMultiplier={15}
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "left": 0,
                  "lineHeight": 24,
                  "opacity": 1,
                  "position": "absolute",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
              testID="text-input-label-inactive"
            >
              min
            </Text>
          </View>
          <TextInput
            editable={true}
            keyboardType="numeric"
            maxFontSizeMultiplier={15}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(73, 69, 79, 1)"
            size="md"
            style={
              Array [
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "flexGrow": 1,
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                  "paddingBottom": 8,
                  "paddingTop": 24,
                },
                Object {
                  "height": 56,
                },
                false,
                Object {
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                false,
                undefined,
              ]
            }
            testID="text-input-flat"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "animationScale": 1,
            "backgroundColor": "rgba(231, 224, 236, 1)",
            "borderBottomLeftRadius": 0,
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 4,
            "flexDirection": "row",
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "rgba(73, 69, 79, 1)",
              "bottom": 0,
              "height": 1,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "zIndex": 1,
            }
          }
          testID="text-input-underline"
        />
        <View
          style={
            Object {
              "flexGrow": 1,
              "minHeight": 56,
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <View
            collapsable={false}
            onLayout={[Function]}
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "zIndex": 3,
              }
            }
          >
            <Text
              collapsable={false}
              maxFontSizeMultiplier={15}
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "left": 0,
                  "lineHeight": 24,
                  "opacity": 0,
                  "position": "absolute",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
              testID="text-input-label-active"
            >
              max
            </Text>
            <Text
              collapsable={false}
              maxFontSizeMultiplier={15}
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "left": 0,
                  "lineHeight": 24,
                  "opacity": 1,
                  "position": "absolute",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
              testID="text-input-label-inactive"
            >
              max
            </Text>
          </View>
          <TextInput
            editable={true}
            keyboardType="numeric"
            maxFontSizeMultiplier={15}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(73, 69, 79, 1)"
            size="md"
            style={
              Array [
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "flexGrow": 1,
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                  "paddingBottom": 8,
                  "paddingTop": 24,
                },
                Object {
                  "height": 56,
                },
                false,
                Object {
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                false,
                undefined,
              ]
            }
            testID="text-input-flat"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
